Applications/Spire/Source/OrderImbalanceIndicator/FilteredOrderImbalanceIndicatorModel.cpp

```diff
@@ -17,42 +18,37 @@ std::tuple<boost::signals2::connection,
       const boost::posix_time::ptime& start,
       const boost::posix_time::ptime& end,
       const OrderImbalanceSignal::slot_type& slot) {
-  auto signal = std::make_unique<FilteredOrderImbalanceSignalConnection>(
-    FilteredOrderImbalanceSignalConnection{
-      {OrderImbalanceSignal(), start, end}, {}});
+  auto signal = std::make_unique<OrderImbalanceSignalConnection>(
+    OrderImbalanceSignalConnection{OrderImbalanceSignal(), start, end});
   auto callback = [this, signal = signal.get()] (auto& imbalance) {
```

Comment:
You likely want a ```shared_ptr``` for ```signal```.

Reply:
I think it could work either way. If the model is destroyed then the connection will be disconnected so the callback is never called.